### PR TITLE
Fixing wrapper_heap alignment problems

### PR DESCRIPTION
**Pull request by sithhell**
This patch ensures that the addresses returned from the wrapper_heap are
correctly aligned for the underlying objects.

### CODE REVIEW

**Comment by sithhell** [comment]
This PR should be ready now.

File: src/runtime/components/server/wrapper_heap.cpp [hunk 1]
```diff
@@ -319,25 +329,27 @@ namespace hpx { namespace components { namespace detail
...
+        pool_ = static_cast<char *>(allocator_type::alloc(total_num_bytes));
         if (nullptr == pool_)
         {
             return false;
         }
 
-        first_free_ = pool_;
-        size_ = s / element_size_; //-V104
-        free_size_ = size_;
+        first_free_ = (reinterpret_cast<std::size_t>(pool_)
```
Comment (biddisco) [inline src/runtime/components/server/wrapper_heap.cpp]: uintptr_t maybe?

Comment (sithhell) [reply src/runtime/components/server/wrapper_heap.cpp]: `std::size_t` is good enough as it is guaranteed to be able to hold pointers.

**Comment by sithhell** [comment]
@hkaiser do you still want to test this on windows? Appveyor is fine with it. I'd really like to move on here, the bug that's fixed here is really severe.

**Comment by hkaiser** [comment]
@sithhell I'd appreciate it if I had a bit more time to test this on windows.

**Comment by sithhell** [comment]
```
[22:15:21] <hkaiser> heller: pls go ahead with merging #3007, I won't have the time to thoroughly investigate things right now
[22:15:39] <hkaiser> if it breaks something we'll fix it later
```

**Pull request merged**